scripts/mssql_discover.py

```diff
@@ -0,0 +1,78 @@
+import sqlalchemy
+
+
+# NB. These are connection secrets, never ever commit these
+USER = ""
+PASS = ""
+IP = ""
+PORT = "1433"
+DB = ""
+
+MASTER_MSSQL_URL = f"mssql+pyodbc://{USER}:{PASS}@{IP}:{PORT}/{DB}?driver=ODBC+Driver+17+for+SQL+Server"
+
+
+SUPPORTED_DATA_TYPES = set(
+    [
+        # char types
+        "varchar",
+        "nvarchar",
+        "char",
+        "nchar",
+        "ntext",
+        "text",
+        # numeric types
+        "int",
+        "bigint",
+        "smallint",
+        "tinyint",
+        "money",
+        "float",
+        "decimal",
+        # date types
+        "date",
+        "datetime",
+        "datetime2",
+        "smalldatetime",
+        # other types
+        "bit",
+    ]
+)
+
+
+def mssql_discover():
+    """
+    Select all databases from the instance
+    Select the schema data for each data base
+    Check if there are any fields in the schema that Fidesops does not yet support
+    """
+    engine = sqlalchemy.create_engine(MASTER_MSSQL_URL)
+    all_dbs = engine.execute("SELECT name FROM sys.databases;").all()
+    all_columns = []
+    flagged_columns = []
+    flagged_datatypes = set()
+    for db_name in all_dbs:
+        db_name = db_name[0]
+        try:
+            columns = engine.execute(
+                f"SELECT TABLE_NAME, COLUMN_NAME, DATA_TYPE, IS_NULLABLE FROM {db_name}.INFORMATION_SCHEMA.COLUMNS;"
+            ).all()
+        except Exception:
+            # print(f"Access to {db_name}'s tables denied.")
+            continue
+
+        all_columns.extend(columns)
+        for table, column, data_type, nullable in columns:
+            if data_type not in SUPPORTED_DATA_TYPES:
+                flagged_datatypes.add(data_type)
+                flagged_columns.append(f"{table}.{column}: {data_type}")
+
+    print(f"{len(set(all_columns))} columns found")
```

Comment:
If you want a count of all columns why use a set?  This would be all unique columns rather than all columns.